src/klein/app.py

```diff
@@ -302,7 +304,10 @@ def error_handler(request, failure):
         def deco(f):
             @wraps(f)
             def _f(instance, request, failure):
-                return _call(instance, f, request, failure)
+                r = _call(instance, f, request, failure)
+                if IRenderable.providedBy(r):
```

Comment:
Yes, I think we should look at that, though I'd suggest a separate ticket, as that's potentially a larger impact.  I wasn't sure, for example, that handling resources here was appropriate, so only added code for renderables.

I'll file the other ticket and merge this, since I think this is still an improvement, if an incomplete one.